src/persist/src/unreliable.rs

```diff
@@ -218,53 +218,61 @@ where
 
 /// An unreliable delegate to [BlobMulti].
 #[derive(Debug)]
-pub struct UnreliableBlobMulti<B> {
+pub struct UnreliableBlobMulti {
     rng: tokio::sync::Mutex<SmallRng>,
-    blob: B,
+    should_happen: f64,
+    should_timeout: f64,
+    blob: Arc<dyn BlobMulti + Send + Sync>,
 }
 
-impl<B> UnreliableBlobMulti<B> {
+impl UnreliableBlobMulti {
     /// Returns a new [UnreliableBlobMulti].
     ///
     /// TODO: Once we turn down the old persist API, return an
     /// [UnreliableHandle] with the previous totally available or unavailable
     /// behavior, plus the new partially available behavior
-    pub fn new_from_seed(seed: u64, blob: B) -> Self {
+    ///
+    /// TODO: Ditto the unreliability should be set via [UnreliableHandle].
+    pub fn new_from_seed(
+        seed: u64,
+        unreliability: f64,
```

Comment:
the unreliability parameter makes things a little bit unintuitive for me because for example, I would expect 50% unreliability to mean "50% of the requests either fail or timeout, and 50% of the requests succeed", but instead it  currently means - 50% of the requests fail, 25% of the requests timeout, and 25% of the requests succeed.

Reply:
works for me!